src/notification/email/EMailNotificationTask.ts

```diff
@@ -141,87 +135,142 @@ export default class EMailNotificationTask implements NotificationTask {
-        } catch (error) {}
-        // Retry?
-        if (!retry && this.serverBackup) {
-          return this.sendEmail(email, data, tenant, user, severity, true);
-        }
-      } else {
-        // Email sent successfully
-        Logging.logDebug({
-          tenantID: tenant.id,
-          source: (Utils.objectHasProperty(data, 'chargeBoxID') ? data.chargeBoxID : undefined),
+    });
+    try {
+      // Send the message
+      const messageSent: Message = await this.client.sendAsync(messageToSend);
+      // Email sent successfully
+      Logging.logDebug({
+        tenantID: tenant.id ? tenant.id : Constants.DEFAULT_TENANT,
+        source: Utils.objectHasProperty(data, 'chargeBoxID') && data.chargeBoxID,
+        action: ServerAction.EMAIL_NOTIFICATION,
+        module: MODULE_NAME, method: 'sendEmail',
+        actionOnUser: user,
+        message: `Email Sent: '${rfc2047.decode(messageSent.header.subject)}'`,
+        detailedMessages: [
+          {
+            email: {
+              from: rfc2047.decode(messageSent.header.from.toString()),
+              to: rfc2047.decode(messageSent.header.to.toString()),
+              subject: rfc2047.decode(messageSent.header.subject)
+            },
+          }, {
+            content: email.html
+          }
+        ]
+      });
+    } catch (error) {
+      // Log
+      try {
+        Logging.logError({
+          tenantID: tenant.id ? tenant.id : Constants.DEFAULT_TENANT,
+          source: Utils.objectHasProperty(data, 'chargeBoxID') && data.chargeBoxID,
           action: ServerAction.EMAIL_NOTIFICATION,
-          module: MODULE_NAME, method: 'prepareAndSendEmail',
+          module: MODULE_NAME, method: 'sendEmail',
+          message: `Error Sending Email (${rfc2047.decode(messageToSend.header.from.toString())}): '${rfc2047.decode(messageToSend.header.subject)}'`,
           actionOnUser: user,
-          message: `Email Sent: '${messageToSend.subject}'`,
           detailedMessages: [
             {
               email: {
-                from: messageToSend.from,
-                to: messageToSend.to,
-                subject: messageToSend.subject
+                from: rfc2047.decode(messageToSend.header.from.toString()),
+                to: rfc2047.decode(messageToSend.header.to.toString()),
+                subject: rfc2047.decode(messageToSend.header.subject)
               },
+            }, {
+              error: error.stack
             }, {
               content: email.html
             }
           ]
         });
+      // For Unit Tests only: Tenant is deleted and email is not known thus this Logging statement is always failing with an invalid Tenant
+      // eslint-disable-next-line no-empty
+      } catch (err) { }
+      let sendSmtpError = true;
+      if (error instanceof SMTPError) {
+        const err: SMTPError = error;
+        switch (err.smtp) {
+          case 510:
```

Comment:
Unfortunately SMTP server does not always return number type as error code, as specified in the IETF RFC, it might be a string formatted '5.1.1', it might be '511', etc. I can't know right know which format will be returned because we do not log the SMTP error deserialised. 
So an enum might not fit: having one key with several possible values to match. That data structure definition can wait for real world usage.

Reply:
Ok